src/Server/Event/Listener/CollectionListener.php

```diff
@@ -0,0 +1,498 @@
+<?php
```

Comment:
@nobesnickr This class needs unit tests.  If I can help with those please let me know.